source/Camera/InfoUtilOther/C++Halcon/HalconConnectUsingGenICam/HalconConnectUsingGenICam.cpp

```diff
@@ -0,0 +1,102 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string filename)
+{
+	const HalconCpp::HString filetype{ "ply" };
+	const HalconCpp::HString filenameCstring{ filename.c_str() };
+	const HalconCpp::HString genParamName{ "invert_normals" };
+	const HalconCpp::HString genParamValue{ "false" };
+	model.WriteObjectModel3d(filetype, filenameCstring, genParamName, genParamValue);
+}
+
+
+HalconCpp::HObjectModel3D setColorInObjectModel3D(HalconCpp::HImage RGB, HalconCpp::HObjectModel3D pcHalcon) {
+
+	HalconCpp::HImage R = RGB.AccessChannel(1);
+	HalconCpp::HImage G = RGB.AccessChannel(2);
+	HalconCpp::HImage B = RGB.AccessChannel(3);
+
+	HalconCpp::HRegion domain = R.GetDomain();
+	HalconCpp::HTuple rows, cols;
+	domain.GetRegionPoints(&rows, &cols);
+
+	HalconCpp::HTuple Rtup = R.GetGrayval(rows, cols);
+	HalconCpp::HTuple Gtup = G.GetGrayval(rows, cols);
+	HalconCpp::HTuple Btup = B.GetGrayval(rows, cols);
+
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
+	pcHalcon.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+
+	return pcHalcon;
+}
+
+
+int main()
+{
+	try
+	{
+		std::cout << "Connecting to camera" << std::endl;
```

Comment:
Move this line above Framegrabber.OpenFramegrabber("Ge.... like in https://github.com/zivid/halcon-samples/blob/master/source/Camera/Basic/Capture.hdev